src/rabbit_channel_interceptor_new.erl

```diff
@@ -0,0 +1,67 @@
+-module(rabbit_channel_interceptor_new).
+
+-include("rabbit_framing.hrl").
+-include("rabbit.hrl").
+
+-export([init/1, intercept_in/3]).
+
+-define(OLD_REGISTRY_CLASS, channel_interceptor).
+-define(NEW_REGISTRY_CLASS, channel_interceptor_new).
+
+-ifdef(use_specs).
+
+% TODO: types
+-callback init(any) -> any.
+-callback intercept(any, any, any) -> any.
+-callback applies_to(any) -> any.
+
+-else.
+
+% TODO
+
+-endif.
+
+init(Ch) ->
+  R = (catch begin
+    Mods = lookup_all(?OLD_REGISTRY_CLASS) ++ lookup_all(?NEW_REGISTRY_CLASS),
+    {Hot, Cold} = lists:partition(fun ({_, M}) -> is_hot(M) end, Mods),
```

Comment:
The aim is to avoid applying 'cold' interceptors to 'hot' methods. If there are only 'cold' interceptors then we only ever apply `fun (M, C) -> {M, C} end` to 'hot' methods/content. This is possibly a premature optimization. I'll do another benchmark without it.